src/legacy/core_plugins/metrics/public/components/vis_types/timeseries/vis.js

```diff
@@ -54,12 +57,35 @@ class TimeseriesVisualization extends Component {
     if (!scaledDataFormat || !dateFormat) return val;
     const formatter = createXaxisFormatter(this.getInterval(), scaledDataFormat, dateFormat);
     return formatter(val);
+  };
+
+  componentDidUpdate() {
+    if (this.showToastNotification && this.notificationReason !== this.showToastNotification.reason) {
+      if (this.notification) {
+        toastNotifications.remove(this.notification);
+      }
+
+      this.notificationReason = this.showToastNotification.reason;
+      this.notification = toastNotifications.addDanger({
+        title: this.showToastNotification.title,
+        text: <MarkdownSimple>{this.showToastNotification.reason}</MarkdownSimple>,
+      });
+    }
+
+    if (!this.showToastNotification && this.notification) {
+      toastNotifications.remove(this.notification);
+      this.notificationReason = null;
+      this.notification = null;
+    }
   }
 
   render() {
     const { backgroundColor, model, visData } = this.props;
     const series = _.get(visData, `${model.id}.series`, []);
     let annotations;
+
+    this.showToastNotification = false;
```

Comment:
nit: What about set `showToastNotification` to `null` to "keep" it's type consistent across the code? 
So instead of having a variable that can hold a boolean or an object, we have a variable that can be an object or null.